### ISSUE TITLE
Die Hard example uses outdated style of defining settings

### ISSUE THREAD

**Issue by nchammas** [HypothesisWorks/hypothesis#3875]
This part of [this article][1] I wrote many years back (and which I am now referencing for my own benefit) no longer works:

[1]: https://hypothesis.works/articles/how-not-to-die-hard-with-hypothesis/

```python
with settings(max_examples=2000):
    DieHardTest = DieHardProblem.TestCase
```

On Hypothesis 6.92.9 this yields:

```python
TypeError: 'settings' object does not support the context manager protocol
```

The correct style is now:

```python
DieHardProblem.TestCase.settings = settings(max_examples=2000)
DieHardTest = DieHardProblem.TestCase
```

When I get a chance I will submit a patch. Just wanted to file this before I forgot.

It would also be good if the whole example were run as part of the CI build as a doctest to ensure the example never breaks silently again. But that's an idea for another day.

**Comment by nchammas** [comment]
> It would also be good if the whole example were run as part of the CI build as a doctest to ensure the example never breaks silently again. But that's an idea for another day.

Or not. Just saw https://github.com/HypothesisWorks/hypothesis/pull/1822.

**Comment by Zac-HD** [comment]
Thanks for the report!  

We'll probably want to fix #2943 in the process; the whole website build is a bit of a mess at the moment.

I'm also open to doing something with doctests; the problem before was that checking exact outputs changed too often and it was a pain to update.  But maybe we can just check that doctests snippets do/don't raise an expected error, or build a tool to auto update the outputs?
